svelte-cloudinary/src/routes/+page.svelte

```diff
@@ -1,51 +1,152 @@
 <script lang="ts">
+	import { tick } from 'svelte';
 	import CldImage from '$lib/components/CldImage.svelte';
+	import CodeBlock from './_components/CodeBlock.svelte';
+	import Toggle from './_components/Toggle.svelte';
+	import type { CldPropOptions, CldPropValueType, CldProps } from './types.ts';
+
+	const baseDir = 'images/';
 	const images = [
-		{ fileName: 'woman-headphones' },
 		{ fileName: 'turtle' },
-		{ fileName: 'sneakers' }
+		{ fileName: 'woman-headphones' },
+		{ fileName: 'sneakers' },
+		{ fileName: 'earth' }
 	];
-	let mainImage = 'earth';
+	let selectedImage = 'turtle';
+
+	let cldProps: CldProps = {
+		effects: [],
+		rawTransformations: []
+	};
 
 	const handleSelectImage = (imgIndex: number) => {
-		mainImage = images[imgIndex].fileName;
+		selectedImage = images[imgIndex].fileName;
+	};
+
+	// this is slightly hacky, but by awaiting this promise in the template below
+	// we can force the image to update with new props. I'm hoping this can help
+	// add a loading state too but haven't tried yet - @troyciesco
+	async function loadImage() {
+		tick();
```

Comment:
@matiasfha quick update on where I'm at exploring storybook - commenting here because I was trying to recreate this functionality on L30 and was running into some issues:
1. there's definitely some plusses and minuses to using SB. It does have a nice familiar layout, and we could probably get all the props in there eventually. We could also easily set up conditional controls, i.e. toggle on pixelation and either leave the default value of true or let the user type in an object. The list might just get kind of long.
2. Biggest issue is the reactivity of the effects (and rawTransformations, etc.) array. adding objects to the effects prop doesn't trigger changes because of the setup in the component itself. 

<img width="1031" alt="Screenshot 2023-07-19 at 1 20 01 PM" src="https://github.com/cloudinary-community/svelte-cloudinary/assets/24758859/8e329fd7-6c02-4f1b-85c2-6781dca89884">

I don't actually think it's _worth_ making effects and other object/array props reactive like src etc. in the main component, since those can be tricky to manage and I don't think most people using this in a normal application will need that (ultimately up to you, of course!). But to get around that on `+page.svelte`, I was running this tick function whenever a prop was changed, which I think is causing the component to rerender appropriately.

I've been tinkering with trying to get that to work in SB with decorators, although I'm not positive it will work - Svelte docs for SB are sparse.

3. another thing to consider is we'll have a little bit less say over how controls work, either not being able to do custom stuff or needing to figure out the SB way. That might be ok, but an example is below - on +page.svelte I'd make text inputs have a debounce to prevent broken fetch requests. Maybe it's possible with SB too, but I'm not sure.

<img width="1418" alt="Screenshot 2023-07-19 at 2 41 20 PM" src="https://github.com/cloudinary-community/svelte-cloudinary/assets/24758859/7a20d518-add6-44a7-b708-b661cda1185b">

I'll tinker a little bit more, but that's what I've found so far!

Reply:
I think, in the meantime we can use the playground you made. It looks good and works as expected.
Great work, thanks for it

Reply:
Sounds good! I'll finish up the cleanup in the other comments, then.